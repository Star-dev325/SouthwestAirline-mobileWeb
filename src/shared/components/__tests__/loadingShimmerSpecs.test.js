import React from 'react';
import LoadingShimmer from 'src/shared/components/loadingShimmer';
import { mount } from 'enzyme';

describe('LoadingShimmer', () => {
  const childWithSize = <div id="test-div" style={{ height: '1px', width: '1px' }} />;

  describe('when shouldDisplay is true', () => {
    it('should render with height and width', () => {
      const wrapper = createComponent({ styles: { height: '1px', width: '1px' } });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when shouldDisplay is false', () => {
    it('should not render animate and effect elements', () => {
      const wrapper = createComponent({ shouldDisplay: false }, childWithSize);

      expect(wrapper).toMatchSnapshot();
    });
  });

  const createComponent = (props = {}, children = <div />) => {
    const defaultProps = { shouldDisplay: true };
    const finalProps = { ...defaultProps, ...props };

    return mount(<LoadingShimmer {...finalProps}>{children}</LoadingShimmer>);
  };
});
