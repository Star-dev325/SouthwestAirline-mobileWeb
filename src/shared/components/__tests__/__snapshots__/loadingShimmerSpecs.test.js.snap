// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingShimmer when shouldDisplay is false should not render animate and effect elements 1`] = `
<LoadingShimmer
  shouldDisplay={false}
>
  <div
    className="shimmer"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <div
      className="shimmer--children"
    >
      <div
        id="test-div"
        style={
          Object {
            "height": "1px",
            "width": "1px",
          }
        }
      />
    </div>
  </div>
</LoadingShimmer>
`;

exports[`LoadingShimmer when shouldDisplay is true should render with height and width 1`] = `
<LoadingShimmer
  shouldDisplay={true}
  styles={
    Object {
      "height": "1px",
      "width": "1px",
    }
  }
>
  <div
    className="shimmer bggray2"
    style={
      Object {
        "height": "1px",
        "width": "1px",
      }
    }
  >
    <div
      className="shimmer--animate"
    >
      <div
        className="shimmer--effect"
      />
      <div
        className="shimmer--effect"
      />
    </div>
    <div
      className="shimmer--children"
    >
      <div />
    </div>
  </div>
</LoadingShimmer>
`;
